### PR TITLE
Added BrowserWindow configuration options support

### PR DESCRIPTION
**Pull request by opudalo**
Added support for [BrowserWindow options](https://github.com/atom/electron/blob/master/docs/api/browser-window.md#new-browserwindowoptions)

Particular use-case - disable web-security for ajax calls testing.

### CODE REVIEW

File: lib/runner.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ var BrowserWindow = require('browser-window');
 var ipc = require('electron').ipcMain;
 var join = require('path').join;
 var resolve = require('path').resolve;
+var extend = require('./extend.js');
```
Comment (juliangruber) [inline lib/runner.js]: i'd rather use this node module: https://www.npmjs.com/package/xtend

File: lib/runner.js [hunk 2]
```diff
@@ -12,11 +13,13 @@ app.on('ready', function(){
...
-          show: opts.show || false,
+        var winOpts = extend({
+          show: false,
           preload: preload,
-          'node-integration': opts['node-integration'] || false
-        });
+          'node-integration': false
+        }, opts)
+
+        win = new BrowserWindow(winOpts)
```
Comment (juliangruber) [inline lib/runner.js]: can be simplified to:

```js
win = new BrowserWindow(extend({
  // ...
}))
```

File: test.js [hunk 3]
```diff
@@ -119,3 +118,16 @@ test('node integraetion', function(t){
...
+    height: 500
+  });
+
+  browser.stdall.on('data', function(l){
+    t.equal(l.toString(), '500\n');
+    t.end();
+  });
+
+  browser.end('console.log(window.outerHeight); window.close()');
+});
```
Comment (juliangruber) [inline test.js]: really well written test!

**Comment by juliangruber** [comment]
..otherwise, +1!

**Comment by opudalo** [comment]
@juliangruber ready to merge whenever you're ready.

**Comment by juliangruber** [comment]
can you add simple documentation as well?

**Comment by opudalo** [comment]
@juliangruber `node-integration`option is from an [old](https://github.com/atom/electron/blob/000625e23f536ee1709ac062aa17d80402f02345/atom/browser/web_contents_preferences.cc#L77) electron API. Now it's `webPreferences.nodeIntegration`

**Comment by opudalo** [comment]
up

**Comment by juliangruber** [comment]
nodeintegration needs to be off still

**Comment by fregante** [comment]
Can this PR be recovered and finished or should it be closed?

**Comment by opudalo** [comment]
@fregante going close it.

**Pull request closed without merging**